vividus/src/test/java/org/vividus/configuration/VaultStoredPropertiesProcessorTests.java

```diff
@@ -50,4 +80,80 @@ void shouldRejectInvalidSecretPaths(String fullSecretPath) throws IOException
+                var annotationConfigApplicationContextConstruction = mockConstruction(
+                        AnnotationConfigApplicationContext.class,
+                        (mock, context) -> {
+                            ConfigurableEnvironment configurableEnvironment = mock();
+                            when(mock.getEnvironment()).thenReturn(configurableEnvironment);
+
+                            when(configurableEnvironment.getPropertySources()).thenReturn(propertySources);
+
+                            VaultTemplate vaultTemplate = mock();
+                            when(mock.getBean(VaultTemplate.class)).thenReturn(vaultTemplate);
+
+                            VaultKeyValueOperations operations = mock();
+                            when(vaultTemplate.opsForKeyValue("secret", KeyValueBackend.KV_2)).thenReturn(operations);
+
+                            var vaultResponse = new VaultResponse();
+                            vaultResponse.setData(Map.of("test", result));
+                            when(operations.get("vividus")).thenReturn(vaultResponse);
+                        }))
+        {
+            var actual = processor.processValue("my-super-password", "secret/vividus/test");
+            assertEquals(result, actual);
+
+            assertThat(annotationConfigApplicationContextConstruction.constructed(), hasSize(1));
+            var context = annotationConfigApplicationContextConstruction.constructed().get(0);
+
+            var ordered = inOrder(context, propertySources);
+            ordered.verify(propertySources).addFirst(
+                    argThat(source -> vaultNamespaceValue.equals(source.getProperty(vaultNamespaceProperty)))
+            );
+
+            ordered.verify(context).registerBean(eq(RestTemplateCustomizer.class), argThat(
+                    (ArgumentMatcher<Supplier<RestTemplateCustomizer>>) supplier -> {
+                        RestTemplate restTemplate = mock();
+                        var interceptors = new ArrayList<ClientHttpRequestInterceptor>();
+                        when(restTemplate.getInterceptors()).thenReturn(interceptors);
+
+                        var restTemplateCustomizer = supplier.get();
+                        restTemplateCustomizer.customize(restTemplate);
+
+                        assertThat(interceptors, hasSize(1));
+                        var interceptor = interceptors.get(0);
+
+                        HttpRequest httpRequest = mock();
+                        var httpHeaders = new HttpHeaders();
+                        when(httpRequest.getHeaders()).thenReturn(httpHeaders);
+
+                        try
+                        {
+                            interceptor.intercept(httpRequest, new byte[0], mock(ClientHttpRequestExecution.class));
+                            assertEquals(Set.of(Map.entry("X-Vault-Namespace", List.of(vaultNamespaceValue))),
+                                    httpHeaders.entrySet());
```

Comment:
in case of assertEquals failure, will we see the original error? just wondering if it will be suppressed by argThat failure

Reply:
AssertionError is shown